### PR TITLE
add repository methods

### PR DESCRIPTION
**Pull request by itkg-nanne**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2032
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/330
https://github.com/open-orchestra/open-orchestra-user-bundle/pull/135
https://github.com/open-orchestra/open-orchestra/pull/1111

### CODE REVIEW

File: UserBundle/Repository/UserRepository.php [hunk 1]
```diff
@@ -181,6 +181,30 @@ public function removeUsers(array $userIds)
...
+     * @param array $groupsId
+     *
+     * @return array
+     */
+    public function countUserByGroup(array $groupsId) {
+        array_walk($groupsId, function(&$item) {$item = new \MongoId($item);});
+        $qa = $this->createAggregationQuery();
+        $qa->project(array('_id' => 0, 'groups' => 1));
+        $qa->unwind('$groups');
+        $qa->group(array('_id' => '$groups', 'tags' => array('$sum' => 1)));
```
Comment (alavieille) [inline UserBundle/Repository/UserRepository.php]: why tags and not sum ?

File: UserBundle/Repository/UserRepository.php [hunk 2]
```diff
@@ -181,6 +181,30 @@ public function removeUsers(array $userIds)
...
+        $qa->match(array('groups.$id' => array('$in' => $groupsId)));
+
+        $aggregateGroupUsers = $qa->getQuery()->aggregate()->toArray();
+        $nbrGroupsUsers = array();
+        array_walk($aggregateGroupUsers, function($item) use (&$nbrGroupsUsers) {
+            $groupId = $item['groups']['$id']->{'$id'};
+            $nbrGroupsUsers[$groupId] = $item['tags'];
+        });
+
+            return $nbrGroupsUsers;
```
Comment (alavieille) [inline UserBundle/Repository/UserRepository.php]: align

File: UserBundle/Repository/UserRepository.php [hunk 3]
```diff
@@ -181,6 +181,30 @@ public function removeUsers(array $userIds)
     }
 
     /**
+     * @param array $groupsId
+     *
+     * @return array
```
Comment (alavieille) [inline UserBundle/Repository/UserRepository.php]: a method count should be return an integer

File: UserBundle/Repository/UserRepository.php [hunk 4]
```diff
@@ -181,6 +181,30 @@ public function removeUsers(array $userIds)
...
+     */
+    public function countUserByGroup(array $groupsId) {
+        array_walk($groupsId, function(&$item) {$item = new \MongoId($item);});
+        $qa = $this->createAggregationQuery();
+        $qa->project(array('_id' => 0, 'groups' => 1));
+        $qa->unwind('$groups');
+        $qa->group(array('_id' => '$groups', 'tags' => array('$sum' => 1)));
+        $qa->project(array('_id' => 0, 'groups' => '$_id', 'tags' => 1));
+        $qa->match(array('groups.$id' => array('$in' => $groupsId)));
```
Comment (alavieille) [inline UserBundle/Repository/UserRepository.php]: you can't match before project  and group ?

Comment (itkg-nanne) [inline UserBundle/Repository/UserRepository.php]: No, it doesn't work.

**Comment by itkg-nanne** [comment]
@alavieille review done

File: UserBundle/Repository/UserRepositoryInterface.php [hunk 5]
```diff
@@ -79,4 +79,11 @@ public function countWithFilterAndSiteIds(PaginateFinderConfiguration $configura
      * @param array $userIds
      */
     public function removeUsers(array $userIds);
+
+    /**
+     * @param array $groupsId
+     *
+     * @return array
+     */
+    public function countsUsersByGroups(array $groupsId);
```
Comment (alavieille) [inline UserBundle/Repository/UserRepositoryInterface.php]: a method count should be return an integer

**Comment by itkg-nanne** [comment]
update done @alavieille

File: UserBundle/Repository/UserRepository.php [hunk 6]
```diff
@@ -181,6 +181,30 @@ public function removeUsers(array $userIds)
     }
 
     /**
+     * @param array $groupsId
+     *
+     * @return array
+     */
+    public function getCountsUsersByGroups(array $groupsId) {
```
Comment (alavieille) [inline UserBundle/Repository/UserRepository.php]: groupIds

File: UserBundle/Repository/UserRepository.php [hunk 7]
```diff
@@ -242,11 +266,8 @@ protected function generateFilterSearch($search, $groupFilter = array())
...
     protected function generateFilterSiteId(array $sitesId)
     {
-        $sitesMongoId = array();
-        foreach ($sitesId as $siteId) {
-            $sitesMongoId[] = new \MongoId($siteId);
-        }
+        array_walk($sitesId, function(&$item) {$item = new \MongoId($item);});
 
-        return array('site.$id' => array('$in' => $sitesMongoId));
+        return array('site.$id' => array('$in' => $sitesId));
```
Comment (alavieille) [inline UserBundle/Repository/UserRepository.php]: $siteIds

File: UserBundle/Repository/UserRepositoryInterface.php [hunk 8]
```diff
@@ -79,4 +79,11 @@ public function countWithFilterAndSiteIds(PaginateFinderConfiguration $configura
      * @param array $userIds
      */
     public function removeUsers(array $userIds);
+
+    /**
+     * @param array $groupsId
+     *
+     * @return array
+     */
+    public function getCountsUsersByGroups(array $groupsId);
```
Comment (alavieille) [inline UserBundle/Repository/UserRepositoryInterface.php]: groupIds

**Pull request merged**